src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -334,10 +336,21 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
   }
 
   override def getUserProxy(userEmail: WorkbenchEmail): Future[Option[WorkbenchEmail]] = {
-    directoryDAO.loadSubjectFromEmail(userEmail).map {
+    directoryDAO.loadSubjectFromEmail(userEmail).flatMap {
       // don't attempt to handle groups or service accounts - just users
-      case Some(user: WorkbenchUserId) => Option(toProxyFromUser(user))
-      case _ => None
+      case Some(user: WorkbenchUserId) => getUserProxy(user)
+      case _ => Future.successful(None)
+    }
+  }
+
+  def getUserProxy(userId: WorkbenchUserId): Future[Option[WorkbenchEmail]] = {
+    directoryDAO.readUserAttribute[String](userId, Attr.proxyEmail).map(_.map(WorkbenchEmail))
+  }
+
+  private def withProxyEmail[T](userId: WorkbenchUserId)(f: WorkbenchEmail => Future[T]): Future[T] = {
+    getUserProxy(userId) flatMap {
+      case Some(e) => f(e)
+      case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"Proxy group does not exist for subject ID: $userId"))
```

Comment:
I think this case can happen both if the user id does not exist or if the proxy address is not present. It seems like the former should be a 404 but the latter is an internal server error. But it seems like we are doing too much error handling... if we get as far as calling this function and the user does not exist, is that an internal server error?

Reply:
Looking at the callers, we can be pretty sure the user exists by the time we get here. Therefore, if the proxy group doesn't exist, that's an unexpected state so I'm good with changing that to an internal server error if that's what you're suggesting.